ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecMapper.java

```diff
@@ -138,6 +143,43 @@ public void configure(JobConf job) {
       }
     }
   }
+
+  /**
+   * defined-self balance ReduceSinkOperator of bucketVersion, keep values to sameness
+   * @param rootOp
+   */
+  private static void balanceRSOpbucketVersion(Operator rootOp){
+    List<Operator<? extends OperatorDesc>> needDealOps = new ArrayList<Operator<? extends OperatorDesc>>();
+    visitChildGetRSOps(rootOp, needDealOps);
+    int bucketVersion = -1;
+    for(Operator<? extends OperatorDesc> rsop : needDealOps){
+      if(rsop.getBucketingVersion() != 2 && rsop.getBucketingVersion() != 1){
```

Comment:
isn't it safe to do it as "and not greater than 2" ?
ideally if we change it to:  if it is zero or -1 use -1 ?

Reply:
can we also see if we can do this check inside the place where the bucketing algorithm is being picked up ?